src/main/java/org/truffleruby/stdlib/psych/PsychParserNodes.java

```diff
@@ -90,442 +73,224 @@
-
-                    } else if (isEvent(event, Event.ID.MappingStart)) {
-                        final MappingStartEvent mappingStartEvent = (MappingStartEvent) event;
-
-                        Object anchor = stringOrNilFor(mappingStartEvent.getAnchor(), tainted, taintNode);
-                        Object tag = stringOrNilFor(mappingStartEvent.getTag(), tainted, taintNode);
-                        Object implicit = mappingStartEvent.getImplicit();
-                        Object style = translateFlowStyle(mappingStartEvent.getFlowStyle());
-
-                        callStartMappingNode.call(frame, handler, "start_mapping", anchor, tag, implicit, style);
-
-                    } else if (isEvent(event, Event.ID.MappingEnd)) {
-                        callEndMappingNode.call(frame, handler, "end_mapping");
-
-                    } else if (isEvent(event, Event.ID.StreamEnd)) {
-                        callEndStreamNode.call(frame, handler, "end_stream");
-                        break;
-                    }
-                }
-            } catch (ParserException | ScannerException pe) {
-                errorProfile.enter();
-                final Mark mark = pe.getProblemMark();
-
-                raiseSyntaxErrorSnippetNode.execute(frame,
-                        "raise Psych::SyntaxError.new(file, line, col, offset, problem, context)",
-                        "file", path,
-                        "line", mark.getLine(),
-                        "col", mark.getColumn(),
-                        "offset", mark.getIndex(),
-                        "problem", pe.getProblem() == null ? nil() : createUTF8String(pe.getProblem()),
-                        "context", pe.getContext() == null ? nil() : createUTF8String(pe.getContext()));
-            } catch (ReaderException re) {
-                errorProfile.enter();
-
-                raiseSyntaxErrorSnippetNode.execute(frame,
-                        "raise Psych::SyntaxError.new(file, line, col, offset, problem, context)",
-                        "file", path,
-                        "line", 0,
-                        "col", 0,
-                        "offset", re.getPosition(),
-                        "problem", re.getName() == null ? nil() : createUTF8String(re.getName()),
-                        "context", toString(re) == null ? nil() : createUTF8String(toString(re)));
-            } catch (Throwable t) {
-                errorProfile.enter();
-                throwException(t);
-                return parserObject;
-            }
-
-            return parserObject;
+    private static int translateFlowStyle(Boolean flowStyle) {
+        if (flowStyle == null) {
+            return STYLE_ANY;
+        } else if (flowStyle) {
+            return STYLE_FLOW;
+        } else {
+            return STYLE_NOT_FLOW;
         }
+    }
 
-        public static void throwException(final Throwable e) {
-            throwsUnchecked(e);
+    @TruffleBoundary
```

Comment:
Do you need a boundary here? `stringFor` already has one.

Reply:
No, indeed.